### DISCUSSION TITLE
Save commonly used Param values as Presets

### DISCUSSION THREAD

**Discussion by matthewblock** [apache/airflow#42192] [Ideas]
Some terminology first since params and param values can get confusing: I'll call a set of values for a DAG's Params a "Configuration". We call it this on the `Trigger DAG w/config` page already. 

Currently when you execute a DAG that has DAG Params, you can select previously used Param values (called "Recent Configurations" in the UI dropdown), as well as the Default parameters, which always appears first in the dropdown:

<img width="1043" alt="image" src="https://github.com/user-attachments/assets/02780c33-c765-4e25-96d8-d886b5a205d2">

It would be useful to pre-define more "Configurations" - Call it Favorites, Presets, or similar - Especially for DAGs that have complex Params or long lists of checkboxes, for example.

The current "best" way to label a "Configuration" is to execute the DAG and provide an optional Run ID:

<img width="825" alt="image" src="https://github.com/user-attachments/assets/6ff9d805-dc90-47dc-9852-9aa2b537b678">

There are a few issues with this:
- You shouldn't necessarily have to run a DAG for a Configuration to exist
- It could be useful to define Configurations in code, alongside the Param defaults
- If you trigger the DAG with a certain Configuration and provide a run ID, the next time you trigger the DAG with that configuration *without* providing a run ID, the labeled run ID configuration goes away (I'm assuming Param values in the list are uniqified)
- The list of Previous Configurations is ordered chronologically by most recent, which isn't desirable if you want Preset Configurations to appear first

Maybe these Preset Configurations could appear in the dropdown after Default parameters but before other Recent Configurations.

This is the reason my team is still using Airflow Variables for complex configurations that can change, alongside custom Pydantic validators and upstream validation task that does the same thing as Params. Right now we copy and paste our config (Airflow variable value) to an IDE and then do find and replace to change our complex configs.

**Comment by Rellum** [comment]
Also looking for this feature.

Also, would be nice to do more complex DAG scheduling with various parameter "presets" (e.g. do frequent smaller incremental DAG runs in addition to less frequent bigger runs).
